scripts/main/photo.js

```diff
@@ -1021,6 +1021,46 @@ photo.getArchive = function (photoIDs, kind = null) {
 	location.href = "api/Photo::getArchive" + lychee.html`?photoIDs=${photoIDs.join()}&kind=${kind}`;
 };
 
+photo.qrCode = function (photoIDs) {
+	if (photoIDs.length === 1) {
+		let myPhoto;
+
+		if (photo.json && photo.json.id === photoIDs[0]) {
+			myPhoto = photo.json;
+		} else {
+			myPhoto = album.getByID(photoIDs[0]);
+		}
+
+		let msg = lychee.html`
+			<div id='qr-code' class='downloads'></div>
+		`;
+
+		basicModal.show({
+			body: msg,
+			callback: function () {
+				QrCreator.render(
+					{
+						text: photo.getViewLink(myPhoto.id),
+						ecLevel: "H",
+						fill: "#000000",
+						background: "#FFFFFF",
+						size: 440, // 500px (modal width) - 2*30px (padding)
+					},
+					document.getElementById("qr-code")
+				);
+			},
+			buttons: {
+				cancel: {
+					title: lychee.locale["CLOSE"],
+					fn: basicModal.close,
+				},
+			},
+		});
+
+		return true;
```

Comment:
Why do you return `true`? If the return value is a constant, I don't see a need for it.

Reply:
https://github.com/LycheeOrg/Lychee-front/pull/287/commits/55fe0cb2375ba5d9c5b25f5db178c189a4e5b6bc